src/confluent_kafka/schema_registry/avro.py

```diff
@@ -133,14 +133,14 @@ class AvroSerializer(Serializer):
     """  # noqa: E501
     __slots__ = ['_hash', '_auto_register', '_known_subjects', '_parsed_schema',
                  '_registry', '_schema', '_schema_id', '_schema_name',
-                 '_subject_name_func', '_to_dict']
+                 '_subject_name_func', '_to_dict', '_named_schemas']
 
     # default configuration
     _default_conf = {'auto.register.schemas': True,
                      'subject.name.strategy': topic_subject_name_strategy}
 
-    def __init__(self, schema_registry_client, schema_str,
-                 to_dict=None, conf=None):
+    def __init__(self, schema_registry_client, schema,
```

Comment:
just looking at this quickly.
can you comment on the pros/cons of passing a `schema` here vs `schema_str` as you see them?

Reply:
Passing a Schema allows setting schema references.